assets/js/modules/analytics/components/dashboard/LegacyAdSenseDashboardWidgetTopPagesTableSmall.js

```diff
@@ -20,95 +20,126 @@
  * WordPress dependencies
  */
 import { __, _x } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import withData from '../../../../components/higherorder/withData';
 import { TYPE_MODULES } from '../../../../components/data';
-import { getDataTableFromData } from '../../../../components/data-table';
+import { STORE_NAME, DATE_RANGE_OFFSET } from '../../datastore/constants';
+import { MODULES_ADSENSE } from '../../../adsense/datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
 import PreviewTable from '../../../../components/PreviewTable';
 import Layout from '../../../../components/layout/Layout';
 import AdSenseLinkCTA from '../common/AdSenseLinkCTA';
-import { getTimeInSeconds } from '../../../../util';
 import {
 	analyticsAdsenseReportDataDefaults,
 	isDataZeroForReporting,
 } from '../../util';
+import { generateDateRangeArgs } from '../../util/report-date-range-args';
+import { getTimeInSeconds, numFmt } from '../../../../util';
+import { getCurrencyFormat } from '../../../adsense/util/currency';
 import TableOverflowContainer from '../../../../components/TableOverflowContainer';
+import ReportTable from '../../../../components/ReportTable';
+import Link from '../../../../components/Link';
+const { useSelect } = Data;
 
-class LegacyAdSenseDashboardWidgetTopPagesTableSmall extends Component {
-	static renderLayout( component ) {
-		return (
-			<div className="
+let currencyFormat;
+const setCurrencyFormat = ( report ) => {
+	currencyFormat = getCurrencyFormat( report );
+};
```

Comment:
This seems really hacky. Why not handle this value relying on hooks in the component as usual?

Reply:
Ah, I hadn't grasped that the `Table` should be inside the component - I was just copying the approach in `LegacyAnalyticsAdSenseDashboardWidgetTopPagesTable`. Good to know about the localisation.
I'll refactor both - no problem 👍